src/sentry/integrations/jira/integration.py

```diff
@@ -559,6 +559,23 @@ def sync_status_outbound(self, external_issue, is_resolved, project_id, **kwargs
 
         client.transition_issue(external_issue.key, transition['id'])
 
+    def _get_done_statuses(self):
+        client = self.get_client()
+        statuses = client.get_valid_statuses()
+        return {
+            s['id'] for s in statuses if s['statusCategory']['key'] == 'done'
+        }
+
+    def should_unresolve(self, data):
+        done_statuses = self._get_done_statuses()
+        return data['changelog']['from'] in done_statuses and \
+            data['changelog']['to'] not in done_statuses
```

Comment:
Nit but I think this would probably be easier to read as 

```py
c1 = data['changelog']['from']
c2 = data['changelog']['to']

return c1 in done_statuses and c2 not in done_statuses
```

super minor nit though